tool/src/main/java/org/wildfly/security/tool/FileSystemEncryptRealmCommand.java

```diff
@@ -0,0 +1,740 @@
+/*
+ * JBoss, Home of Professional Open Source
+ *
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.tool;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.charset.UnsupportedCharsetException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.security.NoSuchAlgorithmException;
+import java.security.Provider;
+import java.security.Security;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+import javax.crypto.SecretKey;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.wildfly.security.WildFlyElytronProvider;
+import org.wildfly.security.auth.realm.FileSystemRealmUtil;
+import org.wildfly.security.auth.realm.FileSystemSecurityRealm;
+import org.wildfly.security.credential.Credential;
+import org.wildfly.security.credential.SecretKeyCredential;
+import org.wildfly.security.credential.store.CredentialStore;
+import org.wildfly.security.credential.store.impl.PropertiesCredentialStore;
+import org.wildfly.security.encryption.SecretKeyUtil;
+import org.wildfly.security.password.WildFlyElytronPasswordProvider;
+import org.wildfly.security.password.spec.Encoding;
+
+/**
+ * Elytron-Tool command to convert un-encrypted FileSystemRealms into an encrypted realm with the use of a SecreKey.
+ * Also, optionally provides a WildFly CLI script to register the FileSystemRealm and corresponding
+ * security-domain in WildFly.
+ * @author <a href="mailto:araskar@redhat.com">Ashpan Raskar</a>
+ */
+
+class FileSystemEncryptRealmCommand extends Command {
+    static final int GENERAL_CONFIGURATION_WARNING = 1;
+    static final String FILE_SYSTEM_ENCRYPT_COMMAND = "filesystem-encrypt";
```

Comment:
Might be good to include realm in the name, e.g., filesystem-realm-encrypt

Reply:
Changed